OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -344,6 +347,45 @@ void onMessageActionOccurredOnPreview(@NonNull final OSInAppMessage message, @No
         logInAppMessagePreviewActions(action);
     }
 
+    /*
+    ** IAM Lifecycle methods
+    ** The following methods call the public inAppMessageLifecycleHandler callbacks
+    */
+
+    void onMessageWillDisplay(@NonNull final OSInAppMessageInternal message) {
+        if (OneSignal.inAppMessageLifecycleHandler == null) {
+            logger.verbose("OSInAppMessageController onMessageWillDisplay: inAppMessageLifecycleHandler is null");
+            return;
+        }
+        OneSignal.inAppMessageLifecycleHandler.onWillDisplayInAppMessage(message);
```

Comment:
We could avoid this static reference to OneSignal class. Can we set the handler directly in this controller? And avoid OneSignal responsibility of having this handler set

Reply:
Should we do the same thing with the IAM click handler while we are at it?

Reply:
Also are we guaranteed to always have the same InAppMessageController for the entire life of the App? If not then we will still need to store the handler in OneSignal statically and inject it into the InAppMessageController upon initialization